### DISCUSSION TITLE
Change to previous bank/bank tree concept

### DISCUSSION THREAD

**Discussion by StylusEcho** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#194] [Feature Requests & Ideas]
Hi, I'm tossing up between Pirate and Morningstar, one feature I noticed on the latter that's missing in the former is the ability to change to the previously visited bank. I believe this would help users a lot in creating menu-like systems for their set-ups.

As an example, you could set up a "utility bank", put all your frequently used things in there, eg tap tempo, quick presets, UI mode change. On your other banks you could switch to the utility bank, do what you need to do, then hit a "back" button to go back to where you were before. 

That's a simple way to do it, but if you want to take it a step further you can arrange specific banks in a folder/tree structure, that would persist even when banks are re-ordered. You could have an "Up" Smart Message to go up the tree and it would have consistent behaviour across banks set up in the structure. This would be good for set lists for instance, or creating comprehensive menu systems for controlling all your pedals.

**Comment by simonaglover95** [comment]
Hi @StylusEcho Sorry for the late reply. We've stopped linking to this dicussion thread, even though it's still available, so we've been monitoring it a lot less. 

You can use Smart Messages on the BRIDGE controllers to do exactly what you've described and many do - not using a "last bank" command, but a "go to bank x" command. If your "menu" is bank 6, just add a "go to bank 6" message to a double press or hold action on all your banks. 

Regarding smart set-list management, we're currently planning some really cool stuff.
